src/runner/runner.go

```diff
@@ -0,0 +1,231 @@
+package runner
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/Arriven/db1000n/src/jobs"
+	"github.com/Arriven/db1000n/src/logs"
+	"github.com/Arriven/db1000n/src/metrics"
+	"github.com/Arriven/db1000n/src/utils"
+)
+
+// Config for the job runner
+type Config struct {
+	ConfigPaths    string        // Comma-separated config location URLs
+	BackupConfig   []byte        // Raw backup config
+	RefreshTimeout time.Duration // How often to refresh config
+	MetricsPath    string        // Where to dump metrics to
+}
+
+// Runner executes jobs according to the (fetched from remote) configuration
+type Runner struct {
+	configPaths    []string
+	backupConfig   []byte
+	refreshTimeout time.Duration
+	metricsPath    string
+
+	currentRawConfig []byte // currently applied config
+
+	log *logs.Logger
+
+	stop chan interface{}
```

Comment:
Also not critical, but it's better to pass `context.Context` instead of reinventing it

Reply:
I don't agree here. `context.Context` is more expensive and all we need here is `chan` to close and signal shutdown. This is a canonical concurrent operation in go.